client/src/app.tsx

```diff
@@ -108,10 +108,10 @@ export function App(appProps) {
   const initialPathname = React.useRef(pathname);
 
   React.useEffect(() => {
-    if (initialPathname.current !== pathname) {
-      setPageNotFound(false);
-    }
-  }, [pathname]);
+    setPageNotFound(
+      appProps.pageNotFound && initialPathname.current === pathname
+    );
+  }, [appProps.pageNotFound, pathname]);
```

Comment:
This looks like a merge artifact that doesn't belong in this PR. 😉

Reply:
I've resolved the issue using interactive rebase. 

When I do rebase in my local environment it always brings in others commits. If someone help me figure out what is going wrong would be great.
I do following steps in my local environment:
```
git checkout main
git pull mdnyari main
git checkout onk-enhance-code-examples
git pull
git rebase main
git push -u origin onk-enhance-code-examples
```

These steps create no issues for mdn/content repo. But for yari I often face this. 🤯